File: LeetCode/455.assign-cookies/assign-cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int gi = g.size() - 1;
        int res = 0;
        for (int si = s.size() - 1; si >= 0 && gi >= 0; si--){
            while (gi >= 0 && g[gi] > s[si])    gi--;
            if (gi >= 0){
                res++;
                gi--;
            }
        }
        return res;
    }
};

File: LeetCode/0455.assign-cookies/assign-cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int res = 0;
        int gi = g.length - 1;
        for (int i = s.length - 1; i >= 0 && gi >= 0; i--){
            while (gi >= 0 && g[gi] > s[i])    gi--;
            if (gi >= 0) {
                res++;
                gi--;
            }
        }
        return res;
    }
}